ui/src/lib/yjs/useYWorkflow.ts

```diff
@@ -139,9 +139,37 @@ export default ({
       handleWorkflowIdChange,
     ],
   );
+
+  const handleWorkflowRename = useCallback(
+    (id: string, name: string) =>
+      undoTrackerActionWrapper(() => {
+        if (!name.trim()) {
+          throw new Error("Workflow name cannot be empty");
+        }
+
+        // Update local state
+        setWorkflows((w) => w.map((w) => (w.id === id ? { ...w, name } : w)));
+
+        // Update subworkflow node in main workflow if this is a subworkflow
+        const mainWorkflow = yWorkflows.get(0);
+        const mainWorkflowNodes = mainWorkflow?.get("nodes") as YNodesArray;
+
+        for (const node of mainWorkflowNodes) {
+          if (node.id === id) {
+            node.data = {
+              ...node.data,
+              name,
+            };
+          }
+        }
+      }),
```

Comment:
This is different from what we discussed I think. Here you are manually modifying the nodes directly, whereas elsewhere we rely on yjs methods by removing and adding nodes again.
I am seeing some performance issues on the PR preview, which might be netlify's fault, but also just to keep updating of yjs state not dissimilar, can you do the same here as the CRUD of nodes and edges? 🙏🏻